resources/web/map/js/trajectory.js

```diff
@@ -0,0 +1,132 @@
+import mapboxGlStyleSwitcher from 'https://cdn.skypack.dev/mapbox-gl-style-switcher';
+
+mapboxgl.accessToken = "pk.eyJ1Ijoic3dpZnQtYWRyaWFuIiwiYSI6ImNsZTN1MW82bDA2OGgzdXFvOWFuZTJlMnEifQ.9nR8m0C-B_ISNR4r4cMExw";
+var map = new mapboxgl.Map({
+    container: 'map',
+    style: "mapbox://styles/mapbox/light-v11",
+    center: [-122.486052, 37.830348],  // Initial focus coordinate
+    zoom: 16
+});
+
+var focus = false;
+
+class FocusToggle {
+    onAdd(map) {
+        this._map = map;
+        this._btn = document.createElement("button");
+        this._btn.className = "mapboxgl-ctrl-icon mapboxgl-ctrl-focus-toggle";
+        this._btn.type = "button";
+        this._btn.onclick = () => {
+            focus = !focus;
+            if (focus) {
+                this._btn.className = "mapboxgl-ctrl-icon mapboxgl-ctrl-unfocus-toggle";
+            } else {
+                this._btn.className = "mapboxgl-ctrl-icon mapboxgl-ctrl-focus-toggle";
+            }
+        };
+        this._container = document.createElement("div");
+        this._container.className = "mapboxgl-ctrl-group mapboxgl-ctrl";
+        this._container.appendChild(this._btn);
+        return this._container;
+    }
+
+    onRemove() {
+        this._container.parentNode.removeChild(this._container);
+        this._map = undefined;
+    }
+}
+
+map.addControl(new mapboxGlStyleSwitcher.MapboxStyleSwitcherControl());
+map.addControl(new FocusToggle(), "top-right");
+map.addControl(new mapboxgl.NavigationControl());
+
+const lines = ["#FF0000", "#FF00FF", "#00FFFF", "#0000FF", "#00FF00", "#000000"];
+
+var data = [];
+
+setupData();
+
+function setupData() {
+    data = [];
+    for (let i = 0; i < lines.length; i++) {
+        data.push({
+            type: 'FeatureCollection',
+            features: []
+        });
+    }
```

Comment:
```suggestion
    data = Array(lines.length).fill({
            type: 'FeatureCollection',
            features: []
        })
```

Reply:
it seems fill creates the same object (breaks colours)

https://stackoverflow.com/a/43461022